notebooks/sycamore-tutorial-intermediate-etl.ipynb

```diff
@@ -0,0 +1,410 @@
+   "metadata": {},
+   "source": [
+    "Let's pause and look at what we've done and what's yet to do. So far, we've read a set of PDF documents from S3, partitioned them with the Aryn Partitioning Service, extracted some metadata from them, chunked them into different size elements, and performed some data cleaning.\n",
+    "\n",
+    "What's left is to embed our documents and write them to Pinecone. First, we'll copy the properties from each document to each element of that document using the `spread_properties` transform. Next, First, we will promote these elements to top-level document objects using the `explode` transform. We do this because Sycamore's `embed` transform only operates on documents, not elements."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "exploded_docset = formatted_docset.spread_properties([\"path\", \"entity\"]).explode()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we'll embed each document with the `embed` transform, and generate term frequency tables for sparse vector search with the `term_frequency` transform. Pinecone requires this latter table."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sycamore.transforms.embed import OpenAIEmbedder\n",
+    "\n",
+    "embedded_docset = (exploded_docset\n",
+    "    .embed(OpenAIEmbedder(model_name=embedding_model))\n",
+    "    .term_frequency(tokenizer=tokenizer, with_token_ids=True)\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, we're ready to write our DocSet to Pinecone using Sycamore's Pinecone connector. Because of lazy execution, this will actually execute the code from all of the cells since the last `show`. If you haven't set your Pinecone API key in your env, you will need to specify it in the options in the `write` transform. Also, if your Pinecone database isn't in `us-east-1`, update the region as well."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pinecone\n",
+    "import time\n",
+    "\n",
+    "start = time.time()\n",
+    "embedded_docset.write.pinecone(\n",
+    "    index_name=\"aryn-etl-tutorial\",\n",
+    "    index_spec=pinecone.ServerlessSpec(cloud=\"aws\", region=\"us-east-1\"),\n",
+    "    dimensions=embedding_dim,\n",
+    "    distance_metric=\"dotproduct\",\n",
+    "    namespace=\"ntsbdocs\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Conclusion and full pipeline \n",
+    "\n",
+    "Congratulations! You just executed an intermediate ETL pipeline with Sycamore and loaded a Pinecone vector database. You partitioned, materialized, extracted, cleaned, chunked, embedded, and loaded your data. \n",
```

Comment:
re: my above comment on intermediate, maybe replace with `comprehensive`

Reply:
Gonna leave it as this, per prior comment